torchao/_models/llama/model.py

```diff
@@ -169,6 +169,12 @@ def setup_caches(self, max_batch_size, max_seq_length, training: bool = False):
 
         self.freqs_cis = precompute_freqs_cis(self.config.block_size, self.config.dim // self.config.n_head, self.config.rope_base, dtype)
         self.causal_mask = torch.tril(torch.ones(self.max_seq_length, self.max_seq_length, dtype=torch.bool))
+    
+    def reset_caches(self):
```

Comment:
is this change needed?

Reply:
Yes, as the `max_batch_size` and `max_seq_length` used during the calibration stage and generation stage may differ, it's necessary to reinitialize the `freqs_cis`.

https://github.com/pytorch/ao/blob/9a0e9f0f26f9982a6a8b38f1998b4affeb63ddc9/torchao/_models/llama/model.py#L156-L160